lib/windows/preplay.py

```diff
@@ -584,6 +589,26 @@ def fillRelated(self, has_prev=False):
         self.relatedListControl.addItems(items)
         return True
 
+    def updateRelated(self):
+        if not self.video.related:
+            return False
+
+        states = {}
+        for rel in self.video.related()[0].items:
+            states[rel.ratingKey] = {
+                "unwatched": not rel.isWatched and '1' or '',
+                "progress": util.getProgressImage(rel)
+            }
+
+        for mli in self.relatedListControl:
```

Comment:
Might we check for new/inexistant related items here and add them accordingly or is ManagedListItem restricted this way?

Reply:
Yeah, but it would probably be easier to just reload the whole thing and update the focused item. Adding and removing items will still upset the focus either way.